src/Flysystem/StreamWrapper.php

```diff
@@ -106,7 +101,7 @@ public function stream_flush(): bool
                 $success = false;
             }
 
-            if (false !== $currentPosition) {
+            if (false !== $currentPosition && is_resource($this->current->handle)) {
```

Comment:
If the handle is not a resource it would have triggered an exception above.

Reply:
It wouldn't, the `writeStream()` call closes the resource. Once it reaches here, it's a closed resource, which doesn't pass `is_resource()` (same issue as with the `stream_close()` above complaining when you pass it a closed resource to close).

Reply:
Could you please provide a test case where this is triggered?